### DISCUSSION TITLE
What's Next

### DISCUSSION THREAD

**Discussion by Zoibderg** [Zoibderg/cs162_group_banking_project#3] [Ideas]
# Project Discussion and Next Steps

Hey team!

I thought it would be a good idea to start a discussion here for sharing ideas and deciding our next steps for the project. Feel free to jump in with suggestions or updates on what you're working on!

## Updates  
To get us started, I’ve quickly thrown together a **Menu Class** as a starting point for our program. Let me know if you want to review or improve it.

## Git & GitHub Resources  
New to Git and GitHub, I found a helpful tutorial:  
[Git & GitHub Crash Course](https://youtu.be/RGOj5yH7evk)

If you’re planning to edit code in your own IDE outside of GitHub, here are the key skills to focus on:
- **Cloning**: Getting the repository onto your local machine.  
- **Pushing**: Sending your changes back to GitHub.  
- **Pulling**: Keeping your local copy up to date with the latest changes.  
- **Committing**: Saving your changes in a meaningful way.  

If you run into any issues or have questions, just let me know—I’m happy to help! 😊

Looking forward to your thoughts and updates!

**Comment by Flareprime** [comment]
I did BankAccountContainer.h and BankAccountContainer.cpp
BankAccountContainer.cpp has functions that should be pretty well documented.

This is very hard to split up, with everything so intertwined. I think we need:
Customer (Customer.h/.cpp) since BankAccountContainer needs it
base BankAccount class (BankAccount.h/.cpp)
derived account classes (SavingsAccount, CheckingAccount, etc.)

**Reply by Zoibderg** [reply]
Implemented Customer in commit 59849d6. Looks like next thing to take is going to be the BankAccount class.

**Reply by 01110100C** [reply]
I can get started on making the derived bank account classes

**Reply by 01110100C** [reply]
have we done anything with the money market yet?

**Reply by Zoibderg** [reply]
> have we done anything with the money market yet?

I haven't touched accounts of any kind yet. Unless Jordan has been working on it, I think everything on here should be up to date.

**Reply by 01110100C** [reply]
okay sounds good. Ill add what I've been working on for the bankaccount class and the derived classes. Still working on it so its not completely finished up but has a basic layout of what it will look like.

**Reply by Flareprime** [reply]
I'm mucking around in the files already written, testing the file input/output stuff
think we need BankAccount before we do all the....accounts right?

**Reply by Flareprime** [reply]
the file is named BankAccount**s**.cpp
Other references might be to BankAccount.cpp 
Gotta pick one, the UML says BankAccount.cpp

also withdrawl = withdrawal

**Reply by 01110100C** [reply]
I was going based off the UML diagram when I was making the bank account and the separate accounts associated with it. I'm getting slightly confused with all the code separated how it is. Are you wanting the savings, checking and credit accounts to be their own files? We can separate them if needed.

**Reply by Zoibderg** [reply]
Yes, I think we should do every class in it's own file that contains only its special functions. Any basic functions will be in the parent class to make the classes more readable and easier to troubleshoot if we run into issues.

**Reply by 01110100C** [reply]
Okay I can separate them when I get home later

**Comment by Flareprime** [comment]
I think every box on the UML represents a class, and in our case, it’s good ideato have each class as its own .cpp and .h file. This way, there are plenty of separate files for the team to work on, minimizing the chances of two people editing the same file at the same time.

For Checking, Savings, Credit, and CD accounts, these should have their own .cpp and .h files since they inherit from BankAccount. These files will be short because most of the methods (like deposit, withdrawal, and balance) are implemented in the BankAccount class.

The BankAccount class is the central hub of this project. BankAccountContainer serves as its "parent" that manages groups of accounts, while Customer and Transaction are heavily intertwined with it. The individual account types (Checking, Savings, etc.) can be thought of as its children that extend its functionality.

I agree that all these intertwined files can make things very confusing, especially when we’re working on separate pieces. On top of that, GitHub is tricky and new.

I can take over BankAccount but won't be done till tomorrow evening at the latest. Nothing can really proceed until its done

**Reply by Zoibderg** [reply]
Let me know if you guys need any help with BankAccount. I'll try and see if I can get the child classes taken care of while you are doing that. I'll follow the UML exactly for now. That way if you guys do the same, the code should match up. Then we can make any adjustments we need after we have everything coded.

**Reply by Flareprime** [reply]
Someone needs to do BankAccount asap so we can finish the rest, getting short on time.

**Reply by Zoibderg** [reply]
I'll have a moment here soon and I'll see if I can blow it out.

**Reply by 01110100C** [reply]
Im at work until 4 today but after that ill be able to get the account classes finished and for the bank account class i did get started on that if you guys want to use what's in that cpp file

**Reply by Flareprime** [reply]
K zoid you...blow... BankAccount, If you blow well, the we can do the rest easily. Let us know your status and if you need aid

**Reply by Zoibderg** [reply]
I've been working on it for that past few hours. I think it's looking good. Taylor had set up a good structure for it, so pretty easy to clean up and split things off. I have the whole day to work on this, so I should have us in a pretty good place by the time you guys are able to jump on tonight.

**Comment by Flareprime** [comment]
my Discord is Flare#1914

**Reply by Zoibderg** [reply]
added

**Reply by Zoibderg** [reply]
@01110100C do you have a discord? That would allow us all to communicate a little easier.

**Reply by 01110100C** [reply]
I don't think I do but I can make one

**Comment by Zoibderg** [comment]
Okay, I just pushed everything that I've been working on all day. I'm a little fried, so there might be some errors, but that should put use in a good place to finish off the Account classes. I'm probably gonna call it a day for now, but let me know if you guys need help with anything. I'll be able to get on late tonight if I need to and I'll also have the day again tomorrow to finish things up.

**Comment by Zoibderg** [comment]
CRUNCH TIME! I spent a lot of time today getting the MoneyMarket class and implementing addAccount functionality into the menu ... lots of little integrations need to be made as our class hierarchy was a little wonky for some reason. 

Right now the main priorities are getting the last CD classes done so that that we can go through and finalize integrating everything in the menu.

**Reply by 01110100C** [reply]
im currently typing up the CD classes I can just start posting what I have if you want me too or I can wait until I have them finished

**Reply by Zoibderg** [reply]
Get in touch with Jordan! I think he is working on those too, that way you guys don't code the same things haha.

**Reply by 01110100C** [reply]
@Flareprime do you already have all three CD classes made?

**Comment by 01110100C** [comment]
@Zoibderg are you submitting the final project?

**Reply by Zoibderg** [reply]
I don't mind who does it

**Reply by 01110100C** [reply]
Okay I don't mind either just wondering cause we are getting close to the finish. Can you just go ahead and submit it please?
